DOCUMENTATION.md

```diff
@@ -107,19 +122,162 @@ In case it's a **dictionary** with fields, please make sure the **encoder** func
 
-### Errors
-Errors can happen on connection for different reasons:
-* Wrong `app_id`, `access_key` or `mqtt_address` were provided to the constructor.
-* The machine may not have access to the network/The MQTT server could be down/Firewall restrictions could prevent connection
-* The client process doesn't have system capabilities to open a socket
-* The MQTT server uses MQTTS, but the client won't accept the TLS certificate
-Errors could also happen when closing connection, in case the disconnection is unexpected.
-It's possible to catch those exceptions using `except RuntimeError as` and print the error.
-
 ### UplinkMessage
 This type of object is constructed dynamically from the message received by the client, so this means some attributes can change from one message to another. However here are some constant attributes usually found in UplinkMessage objects:
-* app_id: the application ID to which the device is registered
-* dev_id: the ID of the device
-* port: the port number on which the message was sent
-* payload_raw: a buffer which contains the payload in hexadecimal
-* metadata: this field is another object which contains all the metadata of the message. Such as: the date, the frequency, the data rate and the list of gateways.
+* `app_id`: the application ID to which the device is registered
+* `dev_id`: the ID of the device
+* `port`: the port number on which the message was sent
+* `payload_raw`: a buffer which contains the payload in hexadecimal
+* `metadata`: this field is another object which contains all the metadata of the message. Such as: the date, the frequency, the data rate and the list of gateways.
+
+
+## ApplicationClient
+The class constructor can be called following this scheme:
+```python
+ApplicationClient(app_id, access_key_or_token, [net_address], [certificate], [discovery_address])
+```
+- `app_id`: **string**  this the name given to the application when it was created.
+![Screenshot of the console with app section](./images/app-console.png?raw=true)
+- `access_key_or_token`: **string**  this can be found at the bottom of the application page under **ACCESS KEYS**. You will need a key allowing you to change the settings if you wish to update your application.
+- `net_address`: **string**  this  is the address of the handler to which the application was registered. It needs to be provided as a `net_address=value` argument when calling the constructor.
+- `certificate`: **string**  this is the content of the certificate used to connect in a secure way to the handler. It needs to be provided as a `certificate=value` argument when calling the constructor.
+- `discovery_address`: **string**  this is the address of the discovery server to use in order to find back the address of the handler to which the application in registered. It needs to be provided as a `discovery_address=value` argument when calling the constructor.
```

Comment:
Same

Reply:
Do you need to indicate the port e.g. `discovery.thethings.network:1900`, or is the hostname enough? Please provide an example for this.